packages/test/test-end-to-end-tests/src/test/upgradeManager.spec.ts

```diff
@@ -43,16 +43,17 @@ describe("UpgradeManager (hot-swap)", () => {
         package: "localLoaderTestPackage",
         config: {},
     };
-    let driver: ITestDriver;
     let opProcessingController: OpProcessingController;
 
     async function createContainer(documentId: string, factory: IFluidDataStoreFactory): Promise<IContainer> {
         const codeLoader: ICodeLoader = new LocalCodeLoader([[codeDetails, factory]]);
+        const driver = getFluidTestDriver();
         const loader = new Loader({
             urlResolver: driver.createUrlResolver(),
             documentServiceFactory: driver.createDocumentServiceFactory(),
             codeLoader,
             options: { hotSwapContext: true },
+            logger: ChildLogger.create(getTestLogger(), undefined, {testDriverType: driver.type}),
```

Comment:
Should we do `?.()` here and for `driver` above? This PR (and maybe existing code) is inconsistent on how defensively these calls are made.